### DISCUSSION TITLE
How would I configure Opportunistic IPSec?

### DISCUSSION THREAD

**Discussion by maglo** [strongswan/strongswan#2095] [Q&A] [answered]
I have the following needs:

- A Host-to-Host IPSec mesh between Windows and Linux nodes (Domain Isolation in Windows parlance)
- Windows to use built-in tools, like Windows Advanced Firewall
- Linux to use any available IPSec implementation, currently evaluating Strongswan, which looks good so far!
- The client Windows computers can exist configured with or without IPSec policies. It is during provisioning that they do not yet have the correct policy.
- The Linux server IPSec policy must hence be able to communicate **both** securely and in the clear with these machines (Opportunistic IPSec)

In Windows, and also in Libreswan, there is a concept which in Libreswan is called Opportunistic IPSec and in Windows it is called Request Security or Require Security. So a server can be configured with Request Security, and if the peer is already configured with IPSec policies, they will be used. If not, communication succeeds anyway.
If the peer is configured with Require Security, it will not communicate with anyone who does not have a corresponding policy.

I have so far managed to configure a swanctl.conf and have IPSec tunnels using between Windows and Linux. Great stuff.

However, I cannot figure out how to configure the Linux server for "Opportunistic IPSec".

Any ideas?

**Comment by tobiasbrunner** [comment, marked as answer]
You can't.

**Reply by maglo** [reply]
thank you for the information
